### PR TITLE
(NETDEV-37) Add hostname provider

### PR DESCRIPTION
**Pull request by willmeek**
This commit adds a hostname provider, with a name string field.

### CODE REVIEW

**Comment by willmeek** [comment]
Customer request. Requires the release of netdev-stdlib and the merging of an accompanying node utils PR

**Comment by willmeek** [comment]
Related node utils PR https://github.com/cisco/cisco-network-node-utils/pull/594

**Comment by mikewiebe** [comment]
@willmeek Make sure to update the changelog.

File: tests/beaker_tests/hostname/hostnamelib.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+###############################################################################
```
[the remark this answers is not in the document]
Comment (willmeek) [reply tests/beaker_tests/hostname/hostnamelib.rb]: Thanks Mike, I have rejigged the tests into the new, cleaner format.

File: tests/beaker_tests/hostname/test_hostname.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
+###############################################################################
```
[the remark this answers is not in the document]
Comment (willmeek) [reply tests/beaker_tests/hostname/test_hostname.rb]: Have rejigged into the new style format.

**Comment by willmeek** [comment]
Added hostname type and provider note to the Changelog under [Unreleased]

**Comment by willmeek** [comment]
Closing this PR temporarily, we have changed the approach to hostname in netdev stdlib, need to rejig the logic here and reopen.

**Comment by willmeek** [comment]
Superceded by https://github.com/cisco/cisco-network-puppet-module/pull/511

**Pull request closed without merging**